lib/ZumoOta/src/Upload.cpp

```diff
@@ -71,57 +74,93 @@ Upload::~Upload()
 void Upload::handleFileUpload(AsyncWebServerRequest *request, const String& filename, size_t index, uint8_t *data, size_t len, bool final)
 {
     String updatedFilename = filename;
+    size_t flashfileSize = 0U; 
+    size_t expectedfileSize = 0U;
 
+    /* Initialize fileSize*/
     if (!filename.startsWith("/"))
     {
         updatedFilename = "/" + filename;
     }
-
-    if (index==0)
+   
+     if (0 == index)
     {   
+        /*Get file size from special header if available*/
+        AsyncWebHeader* headerXFileSizeFirmware = request->getHeader("X-File-Size-Firmware");
+        /* Firmware file size available? */
+        if(nullptr != headerXFileSizeFirmware)
+        {
+            /* If conversion fails, it will contain UPDATE_SIZE_UNKNOWN. */
+            (void)Util::strToUInt32(headerXFileSizeFirmware->value(), expectedfileSize);
+
+           
+            LOG_DEBUG("File size: %d", expectedfileSize);
+            if (expectedfileSize > MAX_FILE_SIZE)
+            {
+                /* If file size is too large, abort upload. */
+                request->send(413, "text/plain", "File size exceeds the limit."); // HTTP-Fehlercode 413 (Payload Too Large)
+                return;
+                LOG_ERROR("File size exceeds the limit.");
+            }
+        }
+        else
+        {
+            LOG_DEBUG("No file size given.");
+        }
+ 
         /*Save file in the request object*/
         request->_tempFile = LittleFS.open(updatedFilename, "w");
         LOG_DEBUG("Upload Start: " + String(updatedFilename));
     }
-    else
-    {
-        LOG_ERROR("Problem to save the request object!");
-       
-    }
+
     if(len)
     {
         /* Write data to the file*/
         request->_tempFile.write(data, len);
     }
 
     /* If this is the last data block, close the file*/
-    if (final)
+    if(final == true)

```

Comment:
before in the list you fixed an if to "Yoda convention" whith constant berfore the "==" operator. So please do it consistently and also here "true == ..."

Reply:
Already adjusted